src/orion/viz/evc/graph/graph.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+"""
+:mod:`orion.viz.evc.graph.graph` -- Build and image graph based on exp-fork tree
+=======================================================================================
+
+.. module:: vizualization
+   :platform: Unix
+   :synopsis: Obtains graph object and outputs an image for research purposes
+              as an image or as a tex based visualization
+
+"""
+import networkx as nx
+import matplotlib.pyplot as plt
+import subprocess
+
+class evc_graph(object):
+    r"""Image graph object
+    
+    Nodes are given as EVC nodes
+
+    Given these EVC nodes we would check if the node is the current root if 
+    we would want to create a the graph of the whole EVC tree. But if not
+    given the root or a specific node, we would want to print out just the 
+    subgraph of the EVC tree.
+    
+    """
+    def __init__(self):
+        super(evc_graph, self, evc_node).__init__()
+        """Initialize the image graph
+
+        """
+
+        '''
+        We assume that when the object get's constructred 
+        an EVC node is passed through.
+        '''
+        self.evc_node = evc_node
+        
+        '''
+        Create a networkx graph and allocate
+        it in memory
+        '''
+        self.graph = nx.Graph()
+
+    def build_graph(self, root=True):
+        """Convert the EVC tree to the networkx graph
+        
+        Algorithm:
+            1. check if root is true
+                1.1 if true call root() on evc node 
+                    to return root
+            2. do a preordertraversal of the tree and
+               build a nn.Graph() 
+
+        """
+
+        if root:
+            self.evc_node = self.evc_node.root()
+        
+        '''
+        Traverse the tree preorderly and then add 
+        EVC node to the nn.Graph() graph
+        '''
+        for node in self.evc_node:
+            # current node is say, root
+            self.graph.add_node(node)
+            # its children connected with the current node
+            # is connected with an edge
+            self.graph.add_edge(node, node.next())
+            # we've built our tree
+        
+        '''
+        Was considering using the map, but giving the node
+        the corresponding edges might not be straightforward
+        '''
+        #self.evc_node.map(self.graph.add_node, self.evc_node)
```

Comment:
@bouthilx is this the right way to ue map? I used an iteration instead of this for now, I want to get it right

Reply:
Nevermind `https://github.com/mila-udem/orion/commit/525bb73ce92d3a40eebd8f3537b24f5c50dd41b0`

Reply:
I think that would work.
```python
def add_node(node, children):
    self.graph.add_node(node)
    for child in children:
        self.graph.add_edge(node, child)

    return node, children

self.evc_node.map(add_node, self.evc_node.children)
```